gst/nnstreamer/tensor_decoder/tensordec.c

```diff
@@ -893,13 +932,57 @@ gst_tensordec_transform_size (GstBaseTransform * trans,
   self = GST_TENSOR_DECODER_CAST (trans);
 
   g_assert (self->configured);
-  g_assert (self->decoder);
 
-  if (self->decoder->getTransformSize)
+  if (!self->is_custom && self->decoder->getTransformSize)
     *othersize = self->decoder->getTransformSize (&self->plugin_data,
         &self->tensor_config, caps, size, othercaps, direction);
   else
     *othersize = 0;
 
   return TRUE;
 }
+
+/**
+ * @brief Registers a callback for tensor_decoder custom condition
+ * @return 0 if success. -ERRNO if error.
+ */
+int
+nnstreamer_decoder_custom_register (const gchar * name,
+    tensor_decoder_custom func, void *data)
+{
+  decoder_custom_cb_s *ptr;
+
+  g_return_val_if_fail (name && strlen (name), -EINVAL);
+  g_return_val_if_fail (func, -EINVAL);
+
+  if (!(ptr = g_try_new0 (decoder_custom_cb_s, 1)))
+    return -ENOMEM;
+
+  ptr->func = func;
+  ptr->data = data;
+
+  if (register_subplugin (NNS_CUSTOM_DECODER, name, ptr) == TRUE)
+    return 0;
+
+  g_free (ptr);
+  return -EINVAL;
+}
+
+/**
+ * @brief Unregisters a callback for tensor_decoder custom condition
+ * @return 0 if success. -ERRNO if error.
+ */
+int
+nnstreamer_decoder_custom_unregister (const gchar * name)
+{
+  decoder_custom_cb_s *ptr;
+
+  ptr = (decoder_custom_cb_s *) get_subplugin (NNS_CUSTOM_DECODER, name);
```

Comment:
Is it okay if we don't do null check on this ptr?

Reply:
I think it would be unnecessary.